src/Microsoft.DotNet.Interactive.Formatting.Tests/HtmlFormatterTests.cs

```diff
@@ -45,13 +45,13 @@ public void Null_references_are_indicated()
             }
 
             [Fact]
-            public void Formatter_does_not_expand_string()
+            public void Formatter_does_not_put_span_around_string()
             {
-                var formatter = HtmlFormatter<string>.GetBestFormatterFor();
+                var formatter = HtmlFormatter.GetBestFormatterFor<string>();
 
                 var s = "hello".ToDisplayString(formatter);
 
-                s.Should().Be("<span>hello</span>");
```

Comment:
This looks like a behavior-breaking change. The `span` element is present to avoid dropping text directly into the DOM.

Reply:
So this is when grabbing the default formater directly.  

However the default "string" formatters were never actually being used (so `span` never actually emitted) because the actual behaviour of "display" was indeed to dump the string straight into the DOM (unencoded), because of this  https://github.com/dotnet/interactive/blob/master/src/Microsoft.DotNet.Interactive.Formatting/Formatter.cs#L265 

So we have three decisions

1. Do we dump strings encoded or not?  I think they have to be encoded or else invalid HTML is produced.  

2. Do we dump with `<span>` or not?  I think not - currently the defulat formatter says "span" but it's never actually emitted today.  It doesn't feel right to start emitting it - if the user wants a span they should make a span element?

3. the separate question of preserving whitespace or not (which we can del with separately)

Reply:
1. We should encode.

2. A user can do `display("hello", "text/html")` and in the absence of this behavior, I believe they'll see no output. My memory is that this bug was the reason for the current behavior.

3. We should preserve whitespace. Perhaps a `pre` tag would be a more informative way to display strings.

Reply:
@jonsequitur I updated the comment above to explain how all three HTML string emits are being aligned to be consistent.

Reply:
> I believe they'll see no output. My memory is that this bug was the reason for the current behavior.

They'll see the output (and note at the moment we're not actually emitting spans - as far as I know that `<span>` is never coming out around strings).  Even in a full document the text "hello" displays as "hello".  I don't know of any situation that where HTML will actually drop encoded text where a span would be valid.

Reply:
> We should preserve whitespace. Perhaps a pre tag would be a more informative way to display strings.

See https://stackoverflow.com/questions/45326130/how-can-i-display-the-text-in-pre-tag-in-the-default-fonts-that-html-uses/45326306 for a possibility (at least for `ToString()` outputs).  This _would_ require a wrapping `<span>` or perhaps even a `<p>`